### PR TITLE
Feature: (ISA-55) Layer preview image improvements

### PR DESCRIPTION
**Pull request by JoshGx2000**
A few fixes to the image generation code, and code for a fallback on the frontend if the image retrieval fails:
Screen Shot 2022-07-27 at 1 27 44 pm

### CODE REVIEW

File: backend/catalogue/management/commands/generate_layer_previews.py [hunk 1]
```diff
@@ -78,42 +92,58 @@ def retrieve_image(layer):
...
+            '--only_generate_missing',
+            help='If true, skips generating images for layers where images already exist'
         )
 
     def handle(self, *args, **options):
-        layer_id = int(options['layer_id']
-                       ) if options['layer_id'] != None else None
+        layer_id = int(options['layer_id']) if options['layer_id'] != None else None
+        only_generate_missing = options['only_generate_missing'].lower() in ['t', 'true'] if options['only_generate_missing'] != None else False
+
```
Comment (markhepburn) [inline backend/catalogue/management/commands/generate_layer_previews.py:106]: Trivial, but there's a tonne of whitespace here!

File: backend/catalogue/management/commands/generate_layer_previews.py [hunk 2]
```diff
@@ -78,42 +92,58 @@ def retrieve_image(layer):
...
+
 
         if layer_id is not None:
             layer = Layer.objects.get(id=layer_id)
 
-            bytes_io = BytesIO()
+            filepath = f'layer_previews/{layer.id}.png'
+            if not only_generate_missing or not default_storage.exists(filepath):
+                bytes_io = BytesIO()
```
Comment (markhepburn) [inline backend/catalogue/management/commands/generate_layer_previews.py:114]: I'd recommend using a with-block for bytesio (just avoids traps with cleanup, etc).  Bit of a dry reference, but these: https://docs.python.org/3/reference/compound_stmts.html#with

**Review by markhepburn — APPROVED** [review]
Looks good -- I'd suggest getting into the habit of using with-blocks in python though (think of them as try-finally; typically just closes resources for you, but safely and idiomatically)

File: backend/catalogue/management/commands/generate_layer_previews.py (hunk 2 above)
Comment (JoshGx2000) [reply backend/catalogue/management/commands/generate_layer_previews.py]: Ah, very useful to know! I guess I should have been more curious about that syntax when it was being used for the database connection. I've used it as suggested.

File: backend/catalogue/management/commands/generate_layer_previews.py (hunk 1 above)
Comment (JoshGx2000) [reply backend/catalogue/management/commands/generate_layer_previews.py:106]: Fixed!

**Pull request merged**